frontend/src/libs/components/loader/loader.tsx

```diff
@@ -1,23 +1,41 @@
 import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { type LoaderShape } from '~/libs/types/types.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
-  children: React.ReactNode;
+  children?: React.ReactNode;
   className?: string;
   isLoading: boolean;
+  hasOverlay?: boolean;
+  loaderType: LoaderShape;
```

Comment:
If that type is used only here - then let's declare it locally for component properties
```suggestion
  loaderType: 'dots' | 'circular';
```